File: logisim/app/src/main/java/logisim/analyze/gui/TruthTablePanel.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.analyze.gui;

import java.awt.Color;
import java.awt.event.MouseEvent;

import logisim.analyze.model.Entry;
import logisim.analyze.model.TruthTable;

interface TruthTablePanel {
	Color ERROR_COLOR = new Color(255, 128, 128);

	TruthTable getTruthTable();

	int getOutputColumn(MouseEvent event);

	int getRow(MouseEvent event);

	void setEntryProvisional(int row, int col, Entry value);
}

File: LogisimPlus/LogisimPlus/logisim/analyze/gui/TruthTablePanel.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.analyze.gui
{
    using LogisimPlus.Java;
    using Entry = logisim.analyze.model.Entry;
	using TruthTable = logisim.analyze.model.TruthTable;

	internal interface TruthTablePanel
	{
		public static Color ERROR_COLOR = Color.FromArgb(255, 255, 128, 128);

		TruthTable TruthTable {get;}

		int getOutputColumn(MouseEvent @event);

		int getRow(MouseEvent @event);

		void setEntryProvisional(int row, int col, Entry value);
	}

}
